tools/backup_analysis.py

```diff
@@ -294,31 +299,38 @@ def print_summary(opts):
     total_files = 0
     total_files_changed = 0
     total_gran_blocks = 0
+    total_single_files = 0
     for t in global_types:
         ts = typestats[t]
         total_chg_bytes += ts.bytes
         total_chg_blocks += ts.chg_blocks
         total_files += ts.files
         total_files_changed += ts.files_changed
         total_gran_blocks += ts.gran_blocks
+        total_single_files += ts.single
     chg_blocks = round(abs(total_chg_blocks / total_gran_blocks * 100))
+    single_pct = round(abs(total_single_files / total_files * 100))
```

Comment:
Is `abs()` needed here? It looks like `total_single_files` will always be positive.

Similar question about the other places here where we `round(abs(...))`.

I guess it's a trade-off between being safe in case something changes in the future vs. having someone read the code and be puzzled about the scenario where these results could be negative.

Just asking the question. If you've considered this and are happy with it this way, that's fine with me.

Reply:
I cleaned this up. It was from before I added code to figure out which backup directory was older/younger. Thanks for asking. It is better without it.